### PR TITLE
Bugfix: mutex_lock acquired for output_frame

### PR DESCRIPTION
**Pull request by siju-samuel**
Calling function 'ActivateNodes' requires holding mutex 'output_frame->mu'

### CODE REVIEW

**Comment by siju-samuel** [comment]
@ymodak could you please assign some reviewers

File: tensorflow/core/common_runtime/executor.cc [hunk 1]
```diff
@@ -2093,8 +2093,11 @@ void ExecutorState::PropagateOutputs(const TaggedNode& tagged_node,
     // Fast path for nodes types that don't need special handling
     DCHECK_EQ(input_frame, output_frame);
     // Normal path for most nodes
+    {
```
Comment (jsimsa) [inline tensorflow/core/common_runtime/executor.cc]: note that on line 2094, we check that `input_frame` and `output_frame` point to the same `FrameState` object

your change make the joint update to `output_frame` and `input_frame` non-atomic ... in other words, it introduces an opportunity for a concurrency error

File: tensorflow/core/common_runtime/executor.cc [hunk 2]
```diff
@@ -2156,6 +2159,7 @@ void ExecutorState::PropagateOutputs(const TaggedNode& tagged_node,
     if (output_frame != nullptr) {
       // This is the case when node is not Enter, Exit, or NextIteration.
       DCHECK(input_frame == output_frame);
+      mutex_lock l(output_frame->mu);
```
Comment (jsimsa) [inline tensorflow/core/common_runtime/executor.cc]: we already hold lock on `input_frame` and since the two alias, this line would deadlock

**Review by jsimsa — CHANGES_REQUESTED** [review]
this PR should not be merged

**Pull request closed without merging**